devices/linux/Files/main.py

```diff
@@ -251,327 +272,369 @@ def prompt_nonempty_string(self, show, prompt, val = ''):
         if Config.verify_user_realm_input:
             if Config.hint_user_input:
-                if self.USERNAME.endswith('@' + Config.user_realm,pos -1):
+                if self.username.endswith('@' + Config.user_realm, pos - 1):
                     debug("realm equal to the expected value")
-                    return(True)
+                    return True
                 else:
                     debug("incorrect realm; expected:" + Config.user_realm)
                     self.alert(Messages.wrong_realm.format(Config.user_realm))
-                    return(False)
-            if self.USERNAME.endswith(Config.user_realm,pos):
+                    return False
+            if self.username.endswith(Config.user_realm, pos):
                 debug("real ends with expected suffix")
-                return(True)
+                return True
             else:
                 debug("realm suffix error; expected: " + Config.user_realm)
-                self.alert(Messages.wrong_realm_suffix.format(Config.user_realm))
-                return(False)
-        pos1 = self.USERNAME.find('@',pos)
+                self.alert(Messages.wrong_realm_suffix.format(
+                    Config.user_realm))
+                return False
+        pos1 = self.username.find('@', pos)
         if pos1 > -1:
             debug("second @ character found")
             self.alert(Messages.wrongUsernameFormat)
-            return(False)
-        pos1 = self.USERNAME.find('.',pos)
+            return False
+        pos1 = self.username.find('.', pos)
         if pos1 == -1:
             debug("no dot in the realm")
             self.alert(Messages.wrongUsernameFormat)
-            return(False)
+            return False
         if pos1 == pos:
             debug("a dot immediately after the @ character")
             self.alert(Messages.wrongUsernameFormat)
-            return(False)
+            return False
         debug("all passed")
-        return(True)
-
-            
+        return True
 
 
 class WpaConf:
-    def prepare_network_block(self,ssid,user_data):
+    """ wpa config class """
+
+    def prepare_network_block(self, ssid, user_data):
+        """ create network config """
         out = """network={
-        ssid=""" + ssid + """
-        key_mgmt=WPA-EAP 
-        pairwise=CCMP  
-        group=CCMP TKIP 
-        eap=""" + Config.eap_outer + """
-        ca_cert=\"""" + os.environ.get('HOME') + """/.cat_installer/ca.pem\"
-        identity=\"""" + user_data.USERNAME + """\"
-        domain_suffix_match=\"""" + Config.server_match + """\"
-        phase2=\"auth=""" + Config.eap_inner + """\"
-        password=\"""" + user_data.PASSWORD + """\"
-        anonymous_identity=\"""" + Config.anonymous_identity + """\"
-} 
-    """
+                 ssid={0}
+                 key_mgmt=WPA-EAP
+                 pairwise=CCMP
+                 group=CCMP TKIP
+                 eap={1}
+                 ca_cert=\"{3}/.cat_installer/ca.pem\"
```

Comment:
I think this has to be a `{2}` ?!

Reply:
Good catch! Fixed with 305b8186dc1dcdf8d2b9363e3a1aecd89ac941fe